web/modules/custom/config_form/src/Form/ConfigForm.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace Drupal\config_form\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Config Form is for the administrator.
+ * 
+ * @package Drupal\config_form\Form
+ */
+class ConfigForm extends ConfigFormBase
+{
+
+  /**
+   * Creating a unique form id for the config form.
+   * 
+   * @return string
+   *   Unique form id.
+   */
+  public function getFormId()
+  {
+    return 'config_form_config';
+  }
+
+  /**
+   * This function is used to tell drupal what type of form it is.
+   * 
+   * @return array
+   *   This array contains the config form.
+   */
+  protected function getEditableConfigNames()
+  {
+    return ['config_form.settings'];
+  }
+
+  /**
+   * Building form for the required fields.
+   * 
+   * @param array $form
+   *   This is the array which will contains fields with associative array.
+   * @param FormStateInterface $form_state
+   *   This variable uses the shows the different form states.
+   * 
+   * @return array
+   *   Form contains the array which having all fields.
+   */
+  public function buildForm(array $form, FormStateInterface $form_state)
+  {
+    $form['full_name'] = [
+      '#type'     => 'textfield',
+      '#title'    => $this->t('Full Name'),
+      '#required' => TRUE,
+    ];
+
+    $form['phone_number'] = [
+      '#type'     => 'tel',
+      '#title'    => $this->t('Phone Number'),
+      '#required' => TRUE,
+    ];
+
+    $form['email'] = [
+      '#type'     => 'email',
+      '#title'    => $this->t('Email'),
+      '#required' => TRUE,
+    ];
+
+    $form['gender'] = [
+      '#type'    => 'radios',
+      '#title'   => $this->t('Gender'),
+      '#options' => [
+        'male'   => $this->t('Male'),
+        'female' => $this->t('Female'),
+        'others' => $this->t('Other'),
+      ],
+      '#required' => TRUE,
+    ];
+
+    $form['action']['submit'] = [
+      '#type'  => 'submit',
+      '#value' => $this->t('Submit')
+    ];
+
+    return $form;
+  }
+
+  /**
+   * This function validate the form input data and checks the conditions.
+   * 
+   * @param array
+   *   Form is the array with a reference.
+   * @param FormStateInterface 
+   *   Holds the values of the input data.
+   * 
+   * @return void
+   *   This function is validating the input data.
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state)
+  {
+    $phone_number = $form_state->getValue('phone_number');
+    $email        = $form_state->getValue('email');
```

Comment:
The logic is correct.
Can we achieve the same using a hook? Considering the form is not a custom one.

Reply:
Yes, it is possible using hook_form_alter() hook.